src/main/scala/com/ubirch/webui/models/keycloak/TokenProcessor.scala

```diff
@@ -99,7 +99,7 @@ object TokenProcessor extends ConfigBase with LazyLogging {
         if (tenantGroup.length > 1) throw new Exception(s"User(${accessToken.getId}) has more than one tenant group. Group paths: ${maybeTenantGroups.mkString(",")}")
         if (tenantGroup.isEmpty) logger.warn(s"User(${accessToken.getId}) doesn't have tenant group. This may cause some problems.")
 
-        tenantGroup.headOption.map(tenantGroup => {
+        Some(tenantGroup.headOption.map(tenantGroup => {
```

Comment:
I think we can improve this expression to make it a bit clearer by using another operator to create the expected option.

Reply:
Sorry I didn't understand. What do you mean by saying 'another operator' ?

Reply:
Like this:

        `
tenantGroup.headOption.map(tenantGroup => {
          val tenant = Util.getRealm(theRealmName).getGroupByPath(tenantGroup).groupRepresentationToTenant

          val subTenants = GroupFactory
            .getById(tenant.id)(theRealmName)
            .toRepresentation
            .getSubGroups
            .asScala
            .map(_.groupRepresentationToTenant)
            .toList

          tenant.copy(subTenants = subTenants)
        }).orElse(Some(GroupFactory.getDefaultTenant))`

Reply:
by using orElse

Reply:
After changing the return type of the `getDefaultTenant` function, this part is became outdated. So after returning 
a Try from `getDefaultTenant` function I don't need to use `getOrElse` or `orElse` anymore